### DISCUSSION TITLE
HTML-like Syntax

### DISCUSSION THREAD

**Discussion by 5no0p** [tetra-framework/tetra#89] [Ideas]
Can you implement HTML-like Syntax?

Django-cotten implement that but they don't have ability to stand alone render like tetra.

**Comment by nerdoc** [comment]
What do you mean exactly with  "implement HTML-like syntax"?
I did not know django-cotton, this seems interesting too. But it has a completely different goal: It just provides something the Tetra `BasicComponent` does, as building block, cotton does this just with a HTML-like syntax, as you seem to recommend.

But Tetra is a "reactive" framework. Components can have backend methods, and attributes that preserve state and react to user events **on the backend**. Django-cotton just provide the frontend block, and replaces that `{% block %}` syntax with a `<c-... />` styled one.

If I didn't miss something when viewing their doc.

So they won't compare, and implementing "HTML like syntax" won't work here much, as Tetra relies on Django's template parser.

I *could* think of a possibility to replace this on the long term with this pseudo-webcomponents syntax in Tetra, but we would loose the ability to benefit from the template compiler, e.g. optimizing things at compile (=server starting) time.

So a "no, we won't do that any time soon" would be the clearest answer here, I'm afraid.
